src/thunderbots/software/ai/main.cpp

```diff
@@ -32,14 +32,32 @@ void worldUpdateCallback(const thunderbots_msgs::World::ConstPtr &msg)
 {
     thunderbots_msgs::World world_msg = *msg;
     World new_world = Util::ROSMessages::createWorldFromROSMessage(world_msg);
-    world.updateBallState(new_world.ball());
-    world.updateFieldGeometry(new_world.field());
-    world.updateEnemyTeamState(new_world.enemyTeam());
-    world.updateFriendlyTeamState(new_world.friendlyTeam());
-    world.updateTimestamp(new_world.getMostRecentTimestamp());
-    RefboxGameState new_game_state =
-        Util::ROSMessages::createGameStateFromROSMessage(world_msg.refbox_data.command);
-    world.updateRefboxGameState(new_game_state);
+    // During RoboCup 2019, we saw weird cases where vision frames wold randomly appear
+    // mirrored. This filter checks if our goalie appears in the enemy defense area to
+    // detect mirroring, and ignores the data if mirroring is detected
+    bool friendly_goalie_in_enemy_defense_area =
+        new_world.friendlyTeam().goalie() &&
+        new_world.field().enemyDefenseArea().containsPoint(
+            new_world.friendlyTeam().goalie()->position());
+    if (!Util::DynamicParameters::AI::vision_flipping_filter_enabled.value() ||
+        !friendly_goalie_in_enemy_defense_area)
```

Comment:
nit; not blocking, but could we change this to `Util::DynamicParameters::AI::vision_flipping_filter_enabled.value() && friendly_goalie_in_enemy_defense_area`? Logically equivalent and easier to follow